tests/integration/test_upgrade.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration test relation file."""
+
+
+import logging
+
+import ops
+import requests
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+from pytest_operator.plugin import OpsTest
+
+import jenkins
+
+from .constants import ALL_PLUGINS
+from .helpers import (
+    generate_client_from_application,
+    get_model_jenkins_unit_address,
+    install_plugins,
+)
+
+LOGGER = logging.getLogger(__name__)
+JENKINS_APP_NAME = "jenkins-k8s-upgrade"
+
+
+async def test_jenkins_deploy_with_plugins(ops_test: OpsTest, model: Model):
```

Comment:
Can you add a blank line to differentiate the arrange, act and assert blocks? https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#test-structure